src/js/tabs/exchange.js

```diff
@@ -54,9 +52,29 @@ ExchangeTab.prototype.angular = function (module)
         exchange.currency_name = currency.to_human({
           full_name: $scope.currencies_all_keyed[currency.get_iso()] ? $scope.currencies_all_keyed[currency.get_iso()].name : null
         });
-        $scope.update_exchange();
+        setImmediate(function() {
+          if ($scope.exchangeForm.amount !== undefined) {
+            $scope.$apply(function() {
+              // hack to re-validate input. remove this and uncomment $validate() when upgraded to angularjs 1.3
```

Comment:
It is already angularjs 1.3 https://github.com/ripple/ripple-client/blob/develop/bower.json#L6